src/Functions/URL/domain.h

```diff
@@ -139,6 +154,8 @@ exloop: if ((scheme_end - pos) > 2 && *pos == ':' && *(pos + 1) == '/' && *(pos
         return std::string_view{};
     if (!has_at_symbol)
         pos = colon_pos ? colon_pos : pos;
+    if (has_open_bracket && has_end_bracket)
```

Comment:
Shouldn't we check something like `has_open_bracket && !has_end_bracket` as well?

Reply:
It is unnecessary. like `has_open_bracket && !has_end_bracket`, due to the presence of `has_open_bracket`,`colon_pos` will always nullptr, the condition` if (has_at_symbol || colon_pos == nullptr) `will always be false, and `dot_pos` will also always be nullptr. When checking and returning the host in `checkAndReturnHost`, it will return an empty string.
I will enhance the test scenarios to cover this case.

Reply:
@Chen768959 could you please comment on that?

Reply:
@vdimir Sure, if `has_open_bracket` (and `!has_end_bracket`),
then `dot_pos` will always be nullptr
(the `logic dot_pos = pos;` for `case : '.'` will never be executed),
so `checkAndReturnHost` will also return `std::string_view{}`. 
So there's no need to check `has_open_bracket && !has_end_bracket`. In this case, it will only return `std::string_view{}`.